bash/sanity_checks.bash

```diff
@@ -299,12 +299,20 @@ function __static__Set_Software_Input_Data_File_If_Not_Set_By_User()
             printf -v filename '%s/%s' \
                 "${HYBRID_software_output_directory[${relative_key}]}" \
                 "${HYBRID_software_default_input_filename[${key}]}"
-        else
+        elif [[ "${filename}" =~ / ]]; then
             if Element_In_Array_Equals_To "${relative_key}" "${HYBRID_given_software_sections[@]}"; then
                 exit_code=${HYBRID_fatal_logic_error} Print_Fatal_And_Exit \
                     'Requesting custom ' --emph "${key}" ' input file although executing ' \
                     --emph "${relative_key}" ' with default output name.'
             fi
+        else
+            if [[ "${filename}" =~ ^('.'|'..')$ ]]; then
+                exit_code=${HYBRID_fatal_wrong_config_file} Print_Fatal_And_Exit \
+                    'Input_File of ' --emph "${key}" ' cannot be ' "${filename}"
+            fi
```

Comment:
Two remarks:
* In Bash, for mind sake, you do not use quotes in a regex. Your line of code is technically correct, but I would bet most of people would be unsure to the question whether the quote is being matched or not. It is way more common to test this like `=~ ^[.][.]?$ ]]`, namely one literal period followed optionally by another one.
* I would `--emph` the filename as well and add a period afterwards.

```bash
            if [[ "${filename}" =~ ^[.][.]?$ ]]; then
                exit_code=${HYBRID_fatal_wrong_config_file} Print_Fatal_And_Exit \
                    'Input_File of ' --emph "${key}" ' cannot be ' --emph "${filename}" '.'
            fi
```